packages/nx/src/project-graph/utils/project-configuration-utils.ts

```diff
@@ -679,6 +682,17 @@ function normalizeTargets(
         delete project.targets[targetName];
       }
     }
+
+    if (
+      disabledTaskSyncGenerators.size &&
+      project.targets[targetName].syncGenerators?.length
+    ) {
+      project.targets[targetName].syncGenerators = project.targets[
+        targetName
+      ].syncGenerators.filter(
+        (syncGenerator) => !disabledTaskSyncGenerators.has(syncGenerator)
+      );
+    }
```

Comment:
Should we keep it in the graph but not run them? :thinking:

Reply:
I considered that, but I don't see the need or much benefit in doing so. The only thing I can think of is that keeping them in the graph would allow us to display them in the PDV but somehow show that they are disabled. I see little value in that, and keeping them in the graph requires handling the `disabledTaskSyncGenerators` option in multiple places.

Could you explain why you think we should keep them in the graph?